src/addrs.ts

```diff
@@ -46,11 +73,13 @@ export function getAddrs(
 
     for (const address of iface) {
       const u8aAddr = ipToU8aAddress(address.address, address.family)
+
+      if (isLinkLocaleAddress(u8aAddr, address.family)) {
+        continue
+      }
+
       if (isPrivateAddress(u8aAddr, address.family)) {
-        if (address.family === 'IPv4' && (options == undefined || options.includeLocalIPv4 != true)) {
-          continue
-        }
-        if (address.family === 'IPv6' && (options == undefined || options.includeLocalIPv6 != true)) {
+        if (address.family === 'IPv4' && (options == undefined || options.includePrivateIPv4 != true)) {
```

Comment:
If no options are provided, skip all ipv4 interfaces?

Reply:
maybe it's better to ignore the case when no options are provided

Reply:
changed, `options` now must not be undefined